### PR TITLE
Remove remaining from_fourier

### PR DESCRIPTION
**Pull request by gkemlin**
There was a remaining "from_fourier" in guess_density.jl. I removed it and created a test case inspired from what lead me to this bug.

### CODE REVIEW

File: test/external_potential.jl [hunk 1]
```diff
@@ -0,0 +1,32 @@
+using DFTK
+using Test
+
+@testset "Externel potential from Fourier coefficients" begin
+    function pot(G)
+        if G == 0
+            return 0.0
```
Comment (antoine-levitt) [inline test/external_potential.jl:7]: zero(G). Also put it on just one line with the ternary operator

File: test/external_potential.jl [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        else
+            1. / (abs(G))
```
Comment (antoine-levitt) [inline test/external_potential.jl:9]: Just 1

File: test/external_potential.jl [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        end
+    end
+
+    a = 10
+    lattice = a .* [[1 0 0.]; [0 0 0]; [0 0 0]];
```
Comment (antoine-levitt) [inline test/external_potential.jl:14]: No need for broadcasting here

File: test/external_potential.jl [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    C = 1.0
+    α = 2;
```
Comment (antoine-levitt) [inline test/external_potential.jl:17]: no ;

File: test/external_potential.jl [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            return 0.0
+        else
+            1. / (abs(G))
+        end
+    end
+
+    a = 10
+    lattice = a .* [[1 0 0.]; [0 0 0]; [0 0 0]];
+
+    C = 1.0
```
Comment (antoine-levitt) [inline test/external_potential.jl:16]: Just 1

**Review by antoine-levitt — COMMENTED** [review]
Nice, thanks!

**Comment by mfherbst** [comment]
Nice. Just shortened the file and added it to the runtests (such that it actually runs).

**Pull request merged**